src/Sylius/Bundle/CoreBundle/EventListener/CustomerDefaultAddressListener.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Sylius\Component\Core\Model\AddressInterface;
+
+class CustomerDefaultAddressListener
+{
+    public function prePersist(LifecycleEventArgs $args)
```

Comment:
I'm not sure, should we base on doctrine event here - this listener would be called each time something is persisted to the database. We can use resource-specific event `sylius.address.pre_create` (`if (!$address instanceof AddressInterface)` is unnecessary then), even though it's also not a perfect solution, but the best in current architecture, I suppose :)

Reply:
Ok, I changed it :+1: